packages/nc-gui/composables/useInfinteData.ts

```
@@ -0,0 +1,1078 @@
+    await Promise.all(
+      rows.map(async (row) => {
+        if (row.rowMeta) {
+          row.rowMeta.changed = false
+          await until(() => !(row.rowMeta?.new && row.rowMeta?.saving)).toMatch((v) => v)
+          row.rowMeta.saving = true
+        }
+      }),
+    )
+
+    const updateArray = rows.map((row) => {
+      const pk = rowPkData(row.row, metaValue?.columns as ColumnType[])
```

Biome

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)